airflow/www/views.py

```diff
@@ -1096,7 +1097,8 @@ def rendered_k8s(self):
         logging.info("Retrieving rendered templates.")
         dag = current_app.dag_bag.get_dag(dag_id)
         task = dag.get_task(task_id)
-        ti = models.TaskInstance(task=task, execution_date=dttm)
+        dag_run = dag.get_dagrun(execution_date=execution_date)
+        ti = dag_run.get_task_instance(task_id=task.task_id)
```

Comment:
At some point we should change these views to use `run_id` right?

Reply:
Yes, that involves changing all the URLs, and that felt like it should be another PR as this one is big enough already :)